From ec60a0b5d86a8b30efbb116231c248e2eaa14468 Mon Sep 17 00:00:00 2001
From: zhang xingxing <2388969553@qq.com>
Date: Sat, 10 Oct 2026 08:33:13 +0200
Subject: [PATCH] avformat/realtextdec: keep the window duration in 64 bits

This is not a security issue.

Fixes: GJrBZmxJqqm8
Fixes: swarm-Integer-truncation-of-parsed-duration-and-timestamps-1a4b12
Found-by: zhang xingxing <2388969553@qq.com>
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
(cherry picked from commit ec60a0b5d86a8b30efbb116231c248e2eaa14468)
---
 libavformat/realtextdec.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libavformat/realtextdec.c b/libavformat/realtextdec.c
index 31f22633412..8f77939210d 100644
--- a/libavformat/realtextdec.c
+++ b/libavformat/realtextdec.c
@@ -64,7 +64,8 @@ static int realtext_read_header(AVFormatContext *s)
     AVStream *st = avformat_new_stream(s, NULL);
     AVBPrint buf;
     char c = 0;
-    int res = 0, duration = read_ts("60"); // default duration is 60 seconds
+    int res = 0;
+    int64_t duration = read_ts("60"); // default duration is 60 seconds
     FFTextReader tr;
     ff_text_init_avio(s, &tr, s->pb);
 
-- 
2.53.0

